docs/design-guides/architecture-overview.md

```diff
@@ -0,0 +1,49 @@
+# Architecture Design
+
+## Table of content
+
+- [Overview](#overview)
+- [Components](#components)
+  - [Motivation for leveraging Graph Open Extensions for Identity Mapping](#motivation-for-leveraging-graph-open-extensions-for-identity-mapping)
+- [Limitations](#limitations)
+- [Guidance](#guidance)
+- [Other Helpful Links to Explore](#other-helpful-links-to-explore)
+
+## Overview
+This sample solution is primarily focused on building the ACS Authentication Backend Server backed by Azure Active Directory as the Identity Provider and using open source libraries from [Microsoft Identity Platform](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-overview). The sample solution can be used directly in below scenarios, otherwise the sample needs to be adapted as described in Guidance section below:
+- The Azure Active Directory is configured to support single tenant use case. 
+> Note: You can verify the configuration through app registration used for user sign in flow for Azure Active Directory instance. Go to the specific app registration within Azure Active Directory through [Azure Portal](https://portal.azure.com/) and check the Authentication tab to verify the tenancy configuration.
+- There are different Active Directory instances configured for different environments. 
+> Note: The sample only supports identity mapping for single environment within Graph Open Extensions used for storage of identity mapping.
+
+## Components
+![Diagram](../images/ACS-Authentication-Server-Sample_Overview-Flow.png)
+
+As seen from the Overview Diagram, the key components of the solution are:
+1. [Secure Web Api backed by Azure Active Directory](./secured-web-api-design.md)
+2. [Identity Mapping leveraging Graph Open Extensions](./Identity-Mapping-Design_Graph-Open-Extensions.md)
+3. Azure Communication Identity service which generates ACS Identity and Tokens. The interaction with ACS Identity service happens within [ACS sdk](https://docs.microsoft.com/en-us/azure/communication-services/concepts/sdk-options#sdks) used within the solution. 
+> Note: The `api/token/teams` endpoint does not leverage #2, as the M365 Azure Active Directory Identity is mapped to user's Teams Identity within ACS Identity Service.
+
+### Motivation for leveraging Graph Open Extensions for Identity Mapping
+The primary motivation comes from the fact that the Azure Communication Identity for the user could be co-located with rest of the information for the Active Active Directory user. And as that would also prevent the usage of additional storage for storing ACS Identity mappings corresponding to Azure Active Directory user Id.
```

Comment:
```suggestion
The Azure Communication Services Identity for the user could be co-located with the information for the Active Active Directory user. This optimizes the complexity to maintain additional storage to keep mappings and instead enables developers to keep everything inside of Azure AD.
```

Reply:
nit: this also has "Active Active Directory" 😄